storybookjs/storybook
# Vue: Pass vite aliases imports to `vue-docgen-api`

Closes #21691 


## What I did
When we use an import alias to a component that `vue-docgen-api` will need to parse to be able to generate the metadata it will not understand how to deal with this import
The import aliases config is declared at the vite, so we need to pass it to `vue-docgen-api` too, but if we pass it directly we will have a problem
In `vite` we can define an absolute path that the reference is actually where the `vite` server is running, example:
```js
{
    "@": "/src"
}
```
If the project is running at the path ```/home/projects/vue-storybook``` `vite` will consider the absolute path ```/src``` relate to the server, but `vue-docgen-api` needs the system absolute path to find the file, in this case the actually path is: ```/home/projects/vue-storybook/src```. So, before passing the aliases to `vue-docgen-api` we need to deal with this kind of absolute paths, we need transform it to actually system paths.


## How to test

It's possible to reproduce the error here:
- [Vue 2 StackBlitz](https://stackblitz.com/edit/github-pnfhmx?file=README.md)
- [Vue 3 StackBlitz](https://stackblitz.com/edit/github-j5c74f?file=README.md)

Or if you want to use the repo:
1. Run a sandbox for template, e.g. `yarn task --task dev --template vue3-vite-default-js`
2. Add in `vite.config` the alias: ```"@": '/src'```
3. Create a mixin inside `src/mixins`
4. Import this mixin using the alias in a component that has a story and use this mixin in the component
5. Open Storybook in your browser
6. Access the story of the component and you will see warning at the terminal



## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
